Failed to parse if let expression

<!--
Thank you for filing a bug report! 🐛 Please provide a short summary of the bug,
along with any information you feel relevant to replicating the bug.
-->

I tried this code: **with -fsyntax-only**

```rust
pub enum Option<T> {
    None,
    Some(T),
}

fn main() {
    let x = Option::Some(3);
    let a = if let Option::Some(1) = x {
        1
    } else if x == Option::Some(2) {
        2
    } else if let Option::Some(y) = x {
        y
    } else {
        -1
    };
}

```

I expected to see this happen: compile without error

Instead, this happened:

```
test.rs:8:16: error: if let expression probably exists, but is being parsed as an if expression. This may be a parser error
    8 |     let a = if let Option::Some(1) = x {
      |                ^~~
test.rs:8:16: error: failed to parse expression in let statement
test.rs:8:5: error: failed to parse statement or expression without block in block expression
    8 |     let a = if let Option::Some(1) = x {
      |     ^~~
test.rs:17:1: error: unrecognised token ‘}’ for start of item
   17 | }
      | ^
test.rs:17:1: error: failed to parse item in crate

```

### Meta

- What version of Rust GCC were you using, git sha if possible.

--- a/gcc/rust/parse/rust-parse-impl.h
+++ b/gcc/rust/parse/rust-parse-impl.h
@@ -13090,7 +13090,7 @@ Parser<ManagedTokenSource>::null_denotation (const_TokenPtr tok,
       return parse_block_expr (std::move (outer_attrs), tok->get_locus ());
     case IF:
       // if or if let, so more lookahead to find out
-      if (lexer.peek_token (1)->get_id () == LET)
+      if (lexer.peek_token ()->get_id () == LET)
 	{
 	  // if let expr
 	  return parse_if_let_expr (std::move (outer_attrs), tok->get_locus ());

--- a/gcc/testsuite/rust/compile/if_let_expr.rs
+++ b/gcc/testsuite/rust/compile/if_let_expr.rs
@@ -1,0 +1,19 @@
+// { dg-options "-fsyntax-only" }
+
+pub enum Option<T> {
+    None,
+    Some(T),
+}
+
+fn main() {
+    let x = Option::Some(3);
+    let a = if let Option::Some(1) = x {
+        1
+    } else if x == Option::Some(2) {
+        2
+    } else if let Option::Some(y) = x {
+        y
+    } else {
+        -1
+    };
+}

